sklearn/utils/estimator_checks.py

```diff
@@ -364,17 +378,19 @@ def check_estimator_sparse_data(name, estimator_orig):
         X = X_csr.asformat(sparse_format)
         # catch deprecation warnings
         with ignore_warnings(category=DeprecationWarning):
+            estimator = clone(estimator_orig)
             if name in ['Scaler', 'StandardScaler']:
-                estimator = clone(estimator).set_params(with_mean=False)
-            else:
-                estimator = clone(estimator)
+                estimator.set_params(with_mean=False)
```

Comment:
I have no idea how to make this into a generizable pattern.

Reply:
I'm not sure any more what I mean (looks at calendar) 18 month ago. Yes, both should pass all tests. I think the point of this might once have been that we check that it does the right thing for sparse data.
And yes, testing both of these would be the right thing. Right now I'm tempted to just remove this statement, and later we can add additional tests.